scipy/interpolate/tests/test_interpolate.py

```diff
@@ -1279,6 +1279,19 @@ def test_integrate(self):
 
         assert_(np.isnan(pp.integrate(a, b, extrapolate=False)).all())
 
+    def test_integrate_readonly(self):
+        x = np.array([1, 2, 4])
+        c = np.array([[0., 0.], [-1., -1.], [2., -0.], [1., 2.]])
+
+        for writeable in (True, False):
+            x.flags.writeable = writeable
+
+            P = PPoly(c, x)
+            I = P.antiderivative()
+
+            vals = I(4) - I(1)
+            assert_(np.isfinite(vals).all())
+
```

Comment:
This does not test the integrate method IIRC.

Reply:
You're right - fixed to call `integrate` directly